packages/tools/client-debugger/client-debugger/src/FluidDebuggerLogger.ts

```diff
@@ -30,7 +90,11 @@ export class FluidDebuggerLogger extends TelemetryLogger {
 		namespace?: string,
 		properties?: ITelemetryLoggerPropertyBags,
 	): TelemetryLogger {
-		return new FluidDebuggerLogger(namespace, properties);
+		// Singleton logger
+		if (FluidDebuggerLogger._instance === undefined) {
+			FluidDebuggerLogger._instance = new FluidDebuggerLogger(namespace, properties);
+		}
```

Comment:
I thought more about this and I think we need to change the approach a bit. Since `create()` could be called with different properties each time, we don't want to return the previous instance if the props won't match what was requested with the new call. So I think the class itself shouldn't be the one ensuring that the same logger object is the one passed around; that should be a responsibility of the application. We can let `create()` return a new instance every time, and instead of having `createFluidContainer()` and `loadExistingFluidContainer()` call it, we can call it once at the root of the module in `client-debugger-view/src/test/app/App.tsx` and pass the returned logger as a new parameter to `createFluidContainer()` and `loadExistingFluidContainer()`, so they can propagate it down. Does that make sense?

Reply:
Make sense. Modified it. Should we allow this logger parameter to be optional, we create one for them if it's missing or required?

Reply:
Let's keep it required for now, since it's code for the test app and we want it to showcase the debugger with all its features. If it's required there's no risk of us forgetting to pass it.